pkg/tfbridge/detailed_diff.go

```diff
@@ -0,0 +1,406 @@
+package tfbridge
+
+import (
+	"context"
+	"errors"
+	"slices"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/walk"
+)
+
+func isPresent(val resource.PropertyValue) bool {
+	return !val.IsNull() &&
+		!(val.IsArray() && val.ArrayValue() == nil) &&
```

Comment:
I'm surprised these cases of degnerate arrays and objects arise, perhaps a bug somewhere?

Reply:
I don't believe anything in the bridge guarantees that the input will be of the right shape, so I've written this a bit defensively.

Specifically, this might come up around schema upgrades and maxitemsone changes